pkg/httphelper/httphelper.go

```diff
@@ -6,14 +6,30 @@ import (
 	"net/http"
 )
 
-func Error(w http.ResponseWriter, err error) {
+type JSONError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Detail  interface{} `json:"detail,omitempty"`
```

Comment:
Yeah, you're right.  We'd end up having to write a factory method and so forth; I'm not a fan scattering map[] work around but in this case for presumably shallow structures, multi-tier serialization won't necessarily come out simplier.